### PR TITLE
TandemFoil: Hard Kutta trailing-edge pressure constraint (from PR #2374)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

PR #2374 showed consistent OOD improvement (-1.5% to -2.1% on p_oodc) by enforcing the Kutta condition as a hard constraint — requiring zero pressure difference at the trailing edge of each airfoil. This is physically grounded: smooth flow at the trailing edge is a fundamental boundary condition for attached flow over airfoils. The current ICML stack has no Kutta enforcement. Adding it as an auxiliary loss term should improve physical consistency, particularly for OOD generalization.

**Note from historical results:** p_oodc improved consistently, but p_tan may regress slightly — monitor both metrics carefully.

## Instructions

1. Read PR #2374 for the full implementation details:
   ```bash
   gh pr view 2374 --json body
   gh pr diff 2374
   ```

2. Port the Kutta constraint loss term into `target/icml2026/train.py`. The constraint penalizes pressure difference between the upper and lower surface at the trailing edge node. Implementation steps:
   - Identify trailing-edge node indices from the geometry (look at how `enable-cp-panel-tandem-only` identifies tandem-specific nodes for reference)
   - Compute `kutta_loss = mean(|p_upper_te - p_lower_te|^2)` over the batch
   - Add `kutta_weight * kutta_loss` to the total training loss

3. Add a `--enable-kutta-constraint` flag (default: False) and `--kutta-weight` (default: 0.1, tunable) to `target/icml2026/train.py`.

4. Run on TandemFoil with the T_max=10 golden config + Kutta enabled:
   ```bash
   cd target/icml2026 && python train.py \
     --dataset tandemfoil \
     --optimizer lion \
     --lr 3e-4 \
     --cosine_t_max 10 \
     --no-use-ema \
     --model_slices 64 \
     --enable-fourier \
     --enable-te-coord-frame \
     --enable-cp-panel \
     --enable-cp-panel-tandem-only \
     --asinh-pressure \
     --residual-prediction \
     --enable-pressure-prior-addition \
     --enable-kutta-constraint \
     --kutta-weight 0.1 \
     --epochs 999 \
     --wandb_group tandem-kutta-constraint
   ```

5. If time allows, try `--kutta-weight 0.01` and `--kutta-weight 0.5` as additional runs in the same W&B group to find the optimal weight.

6. Monitor both `val_primary/surface_pressure_mae` (p_tan) and OOD splits. Historical results showed p_oodc improved but p_tan sometimes regressed — report both clearly.

## Baseline

- **Dataset:** TandemFoilSet
- **Primary metric:** `val_primary/surface_pressure_mae` (lower is better)
- **Current best:** 75.59 (val) / 72.12 (test)
- **Best PR:** #2490 (frieren — T_max=10, Fourier+physics+no-EMA, slices=64, Lion lr=3e-4, 14 epochs)
- **Historical reference (PR #2374):** p_oodc improved -1.5% to -2.1%, but p_tan may regress
- **Baseline W&B run:** 77yoba65
- **Reproduce baseline:**
  ```bash
  cd target/icml2026 && python train.py --dataset tandemfoil --optimizer lion --lr 3e-4 --cosine_t_max 10 --no-use-ema --model_slices 64 --enable-fourier --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --epochs 999
  ```

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2562 because Superseded: historia reassigned to #2619 (DrivAerML WD sweep). NOTE: This was human-directed Kutta TE experiment from issue #2545 — will reassign to idle student..
